sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,543 @@
+    }
+
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+The list of deleted operators are: %, %=, ++, --, +, -, &, |, ^, &=, |=, ^=,
+&&, ||, <<, >>, <<=, >>=, <, >, <=, >=, ~, !
+
+The `make_complex_marray` free function is added to construct complex marrays from real and
+imaginary components. Additionally, the free functions `get_real` and
+`get_imag` are added to access the real and imaginary components of the
+`marray` class without modifying the existing `marray` interface. The usage
+of free functions does cause a deviation from the `std::complex` interface.
+However, it does reduce this extensions impact on the `marray` interface.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+  // Make_complex_marray
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag);
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const T &imag);
+
+  template <class T, std::size_t NumElements>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const T &real, const marray<T, NumElements> &imag);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  marray<sycl::ext::oneapi::complex<T>, NumElements> make_complex_marray(const marray<sycl::ext::oneapi::complex<T>, NumElements> &cmplx, std::integer_sequence<std::size_t, I...> int_seq);
+
+  // Get
+
+  // return marray of component
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  template <class T, std::size_t NumElements>
+  marray<T, NumElements> get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input);
+
+  // return element of component
+  template <class T, std::size_t NumElements>
+  T get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
+
+  template <class T, std::size_t NumElements>
+  T get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::size_t index);
+
+  // return sequence of elements of component
+  template <class T, std::size_t NumElements, std::size_t... I>
+  T get_real(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
+
+  template <class T, std::size_t NumElements, std::size_t... I>
+  T get_imag(const marray<sycl::ext::oneapi::complex<T>, NumElements> &input, std::integer_sequence<std::size_t, I...> int_seq);
```

Comment:
Is the return type of this correct?

Reply:
The return type is wrong. I have updated it to return an marray of the same size as the integer sequence.